src/lib/controllers/configuration-manager/filters.ts

```diff
@@ -0,0 +1,25 @@
+export function escape(text: string): string {
+	return text
+		.split('\n')
+		.join('\\n')
+		.split('\r')
+		.join('');
+}
+
+export async function base64(value: string | Promise<string>): Promise<string> {
+	if (typeof value === 'string') {
```

Comment:
```suggestion
  return Buffer.from((typeof value === 'string') ? value : await value)).toString('base64');
```

Reply:
That would produce a syntax error (extra closing parentheses). Again it's the tradeoff of more readable vs denser. I don't have a strong opinion on this one, so denser syntax it is.